ex4B/ex4.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var tmpStr = '', i = 0, dotRepeat = null;
+
+	// Negative sine can appear only at the start of string and followed by number
+	if (str.slice(i, i + 1) === '-' && isNumber(str.slice(i+1, i + 2))) {
```

Comment:
Using `i` here rather than `0` or `1` doesn't buy you anything. It just makes the code harder to understand. Switch to `0` and `1`.